tests/lib/hardware/libvirt.py

```diff
@@ -79,12 +81,20 @@ def boot(self):
         self._wait_for_ssh()
 
     def destroy(self):
+        # We need to count before destroying the domain
+        data_disk_count = self._data_disk_count()
         self._dom.destroy()
         self._dom.undefine()
         if os.path.exists(self._cloud_init_seed_path):
             os.remove(self._cloud_init_seed_path)
         if os.path.exists(self._snap_img_path):
             os.remove(self._snap_img_path)
+        if data_disk_count > 0:
+            for _id in range(0, data_disk_count):
+                volume_name = f'data-{_id}'
+                disk_path = os.path.join(self._workspace.working_dir,
+                                         f"{self.name}-{volume_name}.qcow2")
```

Comment:
Rather than reconstructing the volume_name and disk_path, why not store a list or dict of volumes as they are created? That way we know exactly what needs cleaning up.

Reply:
Done